tests/foreman/ui/test_ansible.py

```diff
@@ -0,0 +1,117 @@
+"""Test class for Ansible Roles and Variables pages
+
+:Requirement: Ansible
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Ansible
+
+:Assignee: sbible
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from fauxfactory import gen_string
+
+
+def test_positive_import_all_roles(target_sat):
+    """Import all Ansible roles available by default.
+
+    :id: 53fe3857-a08f-493d-93c7-3fed331ed391
+
+    :Steps:
+
+        1. Navigate to the Configure > Roles page.
+        2. Click the `Import from [hostname]` button.
+        3. Get total number of importable roles from pagination.
+        4. Fill the `Select All` checkbox.
+        5. Click the `Submit` button.
+        6. Verify that number of imported roles == number of importable roles from step 3.
+        7. Verify that Ansible variables have been imported along with roles.
+        8. Delete an imported role.
+        9. Verify that the role was successfully deleted.
+
+    :expectedresults: All roles are imported successfully. One role is deleted successfully.
+    """
+    with target_sat.ui_session as session:
+        assert session.ansibleroles.import_all_roles() == session.ansibleroles.imported_roles_count
+        assert int(session.ansiblevariables.read_total_variables()) > 0
+        # The choice of role to be deleted is arbitrary; any of the roles present on Satellite
+        # by default should work here.
+        session.ansibleroles.delete('theforeman.foreman_scap_client')
+        assert not session.ansibleroles.search('theforeman.foreman_scap_client')
+
+
+def test_positive_create_and_delete_variable(target_sat):
+    """Create an Ansible variable with the minimum required values, then delete the variable.
+
+    :id: 7006d7c7-788a-4447-a564-d6b03ec06aaf
+
+    :Steps:
+
+        1. Import Ansible roles if none have been imported yet.
+        2. Create an Ansible variable with only a name and an assigned Ansible role.
+        3. Verify that the Ansible variable has been created.
+        4. Delete the Ansible variable.
+        5. Verify that the Ansible Variable has been deleted.
+
+    :expectedresults: The variable is successfully created and deleted.
+    """
+    key = gen_string('alpha')
+    role = 'redhat.satellite.activation_keys'
+    with target_sat.ui_session as session:
+        if session.ansibleroles.preimport_check() is False:
```

Comment:
```suggestion
        if not session.ansibleroles.preimport_check():
```

Reply:
Actually you can't do that because this function is pretty broken:
https://github.com/SatelliteQE/airgun/blob/master/airgun/entities/ansible_role.py#L54

It returns None or False, which is pretty ambiguous.  Maybe I should have caught that in the airgun review, we were focusing on bigger issues though.

Acceptable as is, but I'd like you to follow this up with a small PR to rename `preimport_check` to something that makes it clear what's being checked and what is being returned. Something like `AnsibleRolesEntity.roles_exist`.  Better yet, don't have the function at all, and use the integer returned by `imported_roles_count`. That's falsy when `0`, and you could use it's return in the same way here. It would be clear what's being checked. You'll need to verify `view.total_imported_roles.read()` doesn't blow up when the page is empty.

Difficult to keep track of how these small pieces fit together when defining new entities and views from scratch.

Reply:
@mshriver I just tried your suggestion of using the integer returned by `imported_roles_count` for this check. Unfortunately, it does indeed blow up if the page is empty. Prior to importing roles, no table is present on the Ansible Roles page. That's why the `preimport_check` function evaluates against `view.pagination.is_displayed` - if there's no pagination displayed, then no roles have been imported.

With this context in mind, would renaming the `preimport_check` function to something more descriptive address your comment sufficiently?

Reply:
I think the imported_roles_count function needs to be updated to not raise exception, but return 0, when there's no table.

Then here you can just directly use `if not imported_roles_count:` to trigger an import.

Reply:
@mshriver Ack, I've just opened an Airgun PR that makes this change: https://github.com/SatelliteQE/airgun/pull/726